src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -41,53 +40,100 @@ export interface CytoscapeMouseOutEvent extends CytoscapeBaseEvent {}
 
 // @todo: Move this class to 'containers' folder -- but it effects many other things
 // exporting this class for testing
-export class CytoscapeLayout extends React.Component<CytoscapeLayoutProps, CytoscapeLayoutState> {
+export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, CytoscapeGraphState> {
   static contextTypes = {
     router: PropTypes.object
   };
 
-  cy: any;
-  graphHighlighter: GraphHighlighter;
+  private graphHighlighter: GraphHighlighter;
+  private cytoscapeReactWrapperRef: any;
+  private newLayout: any;
 
-  constructor(props: CytoscapeLayoutProps) {
+  constructor(props: CytoscapeGraphProps) {
     super(props);
-
-    console.log(`Starting ServiceGraphPage for namespace: ${this.props.namespace.name}`);
+    this.newLayout = '';
   }
 
   shouldComponentUpdate(nextProps: any, nextState: any) {
+    this.newLayout = this.props.graphLayout !== nextProps.graphLayout ? nextProps.graphLayout : '';
     return (
       this.props.isLoading !== nextProps.isLoading ||
       this.props.graphLayout !== nextProps.graphLayout ||
       this.props.showEdgeLabels !== nextProps.showEdgeLabels ||
       this.props.showNodeLabels !== nextProps.showNodeLabels ||
       this.props.showCircuitBreakers !== nextProps.showCircuitBreakers ||
-      this.props.showRouteRules !== nextProps.showRouteRules
+      this.props.showRouteRules !== nextProps.showRouteRules ||
+      this.props.elements !== nextProps.elements ||
+      this.props.graphLayout !== nextProps.graphLayout
     );
   }
 
-  turnEdgeLabelsTo = (value: boolean) => {
+  componentDidMount() {
+    console.log('CY: mounting graph component');
+    this.cyInitialization(this.getCy());
+  }
+
+  componentDidUpdate() {
+    console.log('CY: updating graph component');
+    this.processGraphUpdate(this.getCy());
+  }
+
+  render() {
+    console.log('CY: rendering graph component');
+
+    return (
+      <div id="cytoscape-container" style={{ marginRight: '25em', height: '100%' }}>
+        <Spinner loading={this.props.isLoading}>
```

Comment:
I think this is the reason of not having the ref in the first didMount, it was "loading", there wasn't any graph yet.